SpatialGDK/Source/SpatialGDK/Private/SpatialView/AuthorityRecord.cpp

```diff
@@ -0,0 +1,57 @@
+#include "SpatialView/AuthorityRecord.h"
+
+namespace SpatialGDK
+{
+
+void AuthorityRecord::SetAuthority(Worker_EntityId EntityId, Worker_ComponentId ComponentId, Worker_Authority Authority)
+{
+	const EntityComponentId Id = {EntityId, ComponentId};
+
+	switch (Authority)
+	{
+	case WORKER_AUTHORITY_NOT_AUTHORITATIVE:
+		if (AuthorityGained.RemoveSingle(Id) == 0)
```

Comment:
Why only remove a single entry? And is the order important? If not, `Swap` would be much more efficient.

Reply:
It's effectively a state machine for the change to each entity-component.
One of the invariant held throughout this is that an entity-component can only appear in each list once, and can only appear in one list at a time.

Reply:
Right now the order isn't important so swapping elements would be better. Will add that.